Packages/MIES/MIES_DAEphys.ipf

```diff
@@ -1683,18 +1683,36 @@ Function DAP_SweepRollback(string paneltitle, variable sweepNo, variable newSwee
 	ASSERT(rollbackCountNum == rollbackCountText, "Invalid rollback count")
 End
 
-static Function DAP_UpdateSweepLimitsAndDisplay(panelTitle)
-	string panelTitle
+static Function DAP_UpdateSweepLimitsAndDisplay(string panelTitle, [variable initial])
 
 	string panelList
 	variable sweep, nextSweep, maxNextSweep, numPanels, i
 
 	panelList = GetListofLeaderAndPossFollower(panelTitle)
 
-	if(DAP_DeviceIsLeader(panelTitle))
-		sweep = DAG_GetNumericalValue(panelTitle, "SetVar_Sweep")
+	if(ParamIsDefault(initial))
+		initial = 0
+	else
+		initial = !!initial
+	endif
+
+	if(initial)
+		// we are not implementing sweep adjustment for yoked devices
+		if(!DeviceHasFollower(panelTitle) && !DeviceIsFollower(panelTitle))
+			sweep = AFH_GetLastSweepAcquired(panelTitle) + 1
+			if(IsFinite(sweep))
+				SetSetVariable(panelTitle, "SetVar_Sweep", sweep)
```

Comment:
Why not use PGC_SetAndActivateControl ?

Reply:
Because that will call the GUI control procedure which will think that we are rolling back the sweep, see https://github.com/AllenInstitute/MIES/blob/0b0cf1569a1523aebae2509a0005b35623de7cf1/Packages/MIES/MIES_DAEphys.ipf#L1631. And we don't want to do that.